### PR TITLE
Add Windows Smoke Testing

### PR DESCRIPTION
**Pull request by n1hility**
Add initial podman machine verification testing on Windows. 

A future PR will port the full linux machine testsuite. 

This PR should not be merged until https://github.com/containers/automation_images/pull/228 merges.

```release-note
NONE
```
[NO NEW TESTS NEEDED]

### CODE REVIEW

**Comment by n1hility** [comment]
PTAL @edsantiago @cevich , when you get a chance thanks!

**Comment by edsantiago** [comment]
@n1hility sorry for late response. I've tried looking at this but there's no way I can review it: I have no knowledge of Windows and no way to even test it.

File: contrib/cirrus/win-podman-machine-verify.ps1
Comment (cevich) [inline contrib/cirrus/win-podman-machine-verify.ps1:28]: Minor: IIRC, this is done in another script as well. Is there a way to make/reference some kind of shared function for this to prevent duplication-maintenance headaches?

**Review by cevich — APPROVED** [review]
This is really cool @n1hility 

LGTM

**Comment by cevich** [comment]
@containers/podman-maintainers PTAL

**Comment by n1hility** [comment]
> > I have no knowledge of Windows and no way to even test it.
> 
> Ed, if you take off your glasses, stand on the other side of the room, and squint your eyes real good, it looks _**JUST**_ like bash 🤣

ROFL

File: contrib/cirrus/win-podman-machine-verify.ps1
Comment (n1hility) [reply contrib/cirrus/win-podman-machine-verify.ps1:28]: yes this could be a shared function i can do a followup pr for that

File: .cirrus.yml
Comment (n1hility) [reply .cirrus.yml:1097]: oh sure. I didn't pick up on that order makes sense

File: .cirrus.yml
Comment (edsantiago) [inline .cirrus.yml:1084]: Request: could you please change this to `- alt_build`? `cirrus-task-map` (the script used for generating CI flow diagrams) cannot grok "Windows Cross", and it doesn't look like that's even possible to fix.

File: .cirrus.yml
Comment (n1hility) [reply .cirrus.yml]: moved higher

File: .cirrus.yml
Comment (n1hility) [reply .cirrus.yml:48]: OK Cool. I was hesitant to force a rebuild of all the other images for this, so was a compromise.

**Comment by n1hility** [comment]
/unhold
since image pr has merged

**Comment by n1hility** [comment]
rebased against latest main hoping that resolves the unrelated failures

**Comment by edsantiago** [comment]
Filed #16383 for the (almost certainly a) flake, and restarted test.

**Comment by edsantiago** [comment]
LGTM and CI is finally green, but I'll defer to @cevich for final merge if all his concerns are addressed. Task map is now: 
cirrus-map-pr16257
New task is lavender near lower right.

**Comment by cevich** [comment]
Thanks @n1hility for all your work on this.

**Pull request merged**
